### ISSUE TITLE
get_cpu_flags broken on many CPU models

### ISSUE THREAD

**Issue by startrekdude** [AOMediaCodec/SVT-AV1#1588]
Currently `get_cpu_flags` is implemented like:

```c
if (check_4thgen_intel_core_features()) {
    flags |= CPU_FLAGS_MMX | CPU_FLAGS_SSE | CPU_FLAGS_SSE2 | CPU_FLAGS_SSE3 | CPU_FLAGS_SSSE3 |
             CPU_FLAGS_SSE4_1 | CPU_FLAGS_SSE4_2 | CPU_FLAGS_AVX | CPU_FLAGS_AVX2;
}

if (can_use_intel_avx512()) {
    flags |= CPU_FLAGS_AVX512F | CPU_FLAGS_AVX512DQ | CPU_FLAGS_AVX512CD | CPU_FLAGS_AVX512BW |
              CPU_FLAGS_AVX512VL;
}

return flags;
```

This doesn't always give accurate results because it's possible (and not uncommon) for a CPU to have SSE2, etc, but not necessarily all the way up to AVX2. This causes a very significant performance reduction on certain CPUs.

**Comment by silverbacknet** [comment]
check_4thgen_intel_core_features() *only* checks for full AVX2 support; if that's present then all lower flags are enabled. If not, no flags are enabled at all.

**Comment by hassount** [comment]
Currently, all of our testing and validation resources are using AVX2 capable machines and such is currently stated in the readme file: https://github.com/AOMediaCodec/SVT-AV1#hardware 

That said, this item is open to contributions and we can try to help as much as possible in validating and reviewing the fixes.

**Comment by startrekdude** [comment]
Fixed with #1600
